### PR TITLE
Reduce size of bundled modeler package

### PR DESCRIPTION
**Pull request by JCook21**
This work reduces the size of the bundled modeler package by about 50%. It has been tested in conjunction with changes in the process maker project and may depend on those.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
     "build": "vue-cli-service build",
     "open-cypress": "TZ=UTC cypress open",
     "lint": "vue-cli-service lint --no-fix",
-    "build-bundle": "vue-cli-service build --target lib --name modeler ./src/components/nodes/index.js",
+    "build-bundle": "vue-cli-service build --modern --target lib --name modeler ./src/components/nodes/index.js",
```
Comment (JCook21) [inline package.json]: I'm not totally sure this is doing anything for us yet. I'll dig a little more into this.

File: vue.config.js [hunk 2]
```diff
@@ -26,12 +28,32 @@ module.exports = {
...
+      }
+      return externals;
+    })(),
+    plugins: (() => {
+      const argv = yargs
+        .boolean('analyze')
+        .argv;
+      const plugins = [];
+      if (argv.analyze) {
+        plugins.push(new BundleAnalyzerPlugin());
```
Comment (JCook21) [inline vue.config.js]: This allows us to analyze the bundle size if the `--analyze` flag is passed to the build.

File: vue.config.js [hunk 3]
```diff
@@ -26,12 +28,32 @@ module.exports = {
...
+        externals.push(
+          'vue',
+          /^bootstrap\/.+$/,
+          /^@processmaker\/.+$/,
+          'i18next',
+          '@panter/vue-i18next',
+          'jointjs',
+          'luxon',
+          'bpmn-moddle',
+          'lodash',
```
Comment (JCook21) [inline vue.config.js]: Many of these dependencies were also being included in the process maker project. Including them in modeler was just causing bloat so I moved to exclude them..

File: src/portsUtils.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import { g } from 'jointjs/dist/joint';
+import { g } from 'jointjs';
```
Comment (JCook21) [inline src/portsUtils.js]: Changing this import allows jointjs to be moved to externals.

**Comment by JCook21** [comment]
> I just suggested removing the `modern` flag for now; you can let me know if that makes sense.

Thanks @chipit24, I'll look some more into it. With this option builds give a warning saying that browser list is the better way to do this so I'm not sure...

File: package.json (hunk 1 above)
Comment (chipit24) [reply package.json]: I wrote this comment up but looks like I forgot to include it in the review and it was lost:

I looked into the docs and it seems that in order to take advantage of modern mode, we'd have to include the appropriate `module` and `nomodule` scripts where the app is loaded.

However, when I run the build, I get the following warning:

>  WARN  Modern mode only works with default target (app). For libraries or web components, use the browserslist config to specify target browsers.

And it seems the same files are generated whether or not the `--modern` flag is set. So unless we significantly change our build process (switch from a lib build target to the default app build target), this change may only cause some confusion.

**Pull request merged**